### PR TITLE
Added missing keys in delete chunk and exception

### PR DESCRIPTION
**Pull request by gentlegiantJGC**
A number of keys (including the version, tile entities and entities) were not being deleted when the chunk was deleted making MCedit think the chunk still existed causing further issues with that chunk

Added exception and error message if tile entity data is not in the proper format (this caused a crash without) I believe this was caused by deleting the chunk using the old delete chunk code followed by MCedit thinking the chunk still existed because the version key existed and it wrote the tile entity data back which didn't exist.

Edit: also added the new leveldb dll. This should work with older worlds but I have not tested this.

Fix the code up as required. I don't know all the standard practices here

### CODE REVIEW

File: pymclevel/leveldbpocket.py [hunk 1]
```diff
@@ -1843,7 +1854,11 @@ def add_data(self, terrain=None, tile_entities=None, entities=None, subchunk=Non
             ids_get = self.world.defsIds.mcedit_ids.get
             if DEBUG_PE:
                 write_dump(('\\' * 80) + '\nParsing Entities in chunk %s,%s\n' % (self.chunkPosition[0], self.chunkPosition[1]))
-            Entities = loadNBTCompoundList(entities)
+            try:
+                Entities = loadNBTCompoundList(entities)
+            except:
+                logger.error('potential entity corruption at chunk '+str(self.chunkPosition))
```
Comment (LaChal) [inline pymclevel/leveldbpocket.py]: Please, use the variable interpolation format like on line 1845/1856, or the `format` method of the string object.

**Comment by LaChal** [comment]
Nice addition to the chunk deletion part, indeed.

Regarding the DLL, can you tell us which commit was used from Mojang's repo, and which compiler and options were used?
It's important if we need to recompile on our side.

**Comment by gentlegiantJGC** [comment]
yeh it is probably best you recompile the dll. Neonerz gave me the dll which he got from MCCToolChest as a quick fix so we could continue using MCedit with 1.2

The error I had was when it tried loading the nbt from the database but it wasn't valid nbt. The chunks that were causing the errors were ones I had deleted using the old chunk deletion code. In theory the new code should fix that but it would be good to put in an exception so MCedit doesn't crash if the nbt gets corrupted somehow

File: pymclevel/leveldbpocket.py (hunk 1 above)
Comment (gentlegiantJGC) [reply pymclevel/leveldbpocket.py]: I am aware of this method however I have always done the lazy way. Feel free to reformat if you want

**Comment by LaChal** [comment]
Some issues may be due to the wrong version detection of the world. It was set to 0x05, but 1.2 internal DB version is 0x06. This has been fixed in the commit 47cf3ad.
It solves the chunk not present error, but the entities are still present in the world until it is reloaded.

I tested your code about the deletion, and once the version stuff fixed.
Some improvements are needed but it just deleted the chunk and the other data fine :smile:
It is not implemented int the commit above.

Regarding the DLL, it is really important to know the commit, because we need to build it for Linux (and OSX) too. Without, at least this information, we'll have different executables, which mean different behaviors (and bugs)...

**Comment by gentlegiantJGC** [comment]
In my local version I had disabled the version check and hard coded it to use 1plus so I don't think that was an issue in my case.

Yeh I understand but I don't know. compiling in general is not really something I know a lot about. Ultimatly it will probably need to be recompiled with the latest commit when 1.2 is finally released.

**Comment by gentlegiantJGC** [comment]
@LaChal is this going to get merged with the master? I redownloaded MCedit and deleted chunks and it corrupted chunks and caused mcedit to crash every time it loaded those chunks that were deleted. MCedit seems to write back an almost empty entity key to the database. I think it deletes the data internally but still writes what it has which is nothing

**Comment by LaChal** [comment]
Not yet implemented.

I guess we'll pick the code, optimize it and commit on our side.

As I said, additions to the code are wanted, but not optimized :smile:

**Comment by LaChal** [comment]
Changes implemented.

**Pull request closed without merging**